IPython/core/ultratb.py

```diff
@@ -464,29 +467,38 @@ def _format_list(self, extracted_list):
         Colors = self.Colors
         list = []
         for filename, lineno, name, line in extracted_list[:-1]:
-            item = "  %s in %s%s%s\n" % (
+            item = "  %s" % (
                 _format_filename(
                     filename, Colors.filename, Colors.Normal, lineno=lineno
-                ),
-                Colors.name,
-                name,
-                Colors.Normal,
+                )
             )
+
+            if name != "<module>":
+                item += " in %s%s%s\n" % (
+                    Colors.name,
+                    name,
+                    Colors.Normal,
+                )
+            else:
+                item += "\n"
             if line:
                 item += '    %s\n' % line.strip()
             list.append(item)
         # Emphasize the last entry
         filename, lineno, name, line = extracted_list[-1]
-        item = "%s  %s in %s%s%s%s\n" % (
+        item = "%s  %s" % (
             Colors.normalEm,
             _format_filename(
                 filename, Colors.filenameEm, Colors.normalEm, lineno=lineno
             ),
-            Colors.nameEm,
-            name,
-            Colors.normalEm,
-            Colors.Normal,
         )
+        if name != "<module>":
+            item += " in %s%s%s" % (
```

Comment:
Since we are changing this anyway, can we please get rid of `%`-formatting and use f-strings?

You can leave other lines in the module as is, but let's be good scouts and rice up the ones that you are changing.